src/TConsole.cpp

```diff
@@ -1878,54 +1873,42 @@ std::list<int> TConsole::_getBgColor()
 
 std::list<int> TConsole::getFgColor(std::string& buf)
 {
-    std::list<int> result;
-    std::string key = buf;
-    if (buf == "main") {
+    QString key = QString::fromUtf8(buf.c_str());
+    if (key == QLatin1String("main")) {
         return _getFgColor();
     }
-    if (mSubConsoleMap.find(key) != mSubConsoleMap.end()) {
-        TConsole* pC = mSubConsoleMap[key];
-        if (!pC) {
-            return result;
-        }
+    auto pC = mSubConsoleMap.value(key);
+    if (pC) {
         return pC->_getFgColor();
-    } else {
-        return result;
     }
+
+    return std::list<int>();
```

Comment:
```suggestion
    return {};
```

Less duplication

Reply:
Is that valid? I didn't realise - I just wanted an obviously empty value and I thought that was how it was done - though I guess what you say makes sense if the compiler allows it (is it a C++14 feature or C++11 or even *before*? :astonished: )

Reply:
Yup

Reply:
Ah, [`return {};`](https://en.cppreference.com/w/cpp/language/return) is a **C++11** feature so *is* usable for us in Mudlet - you learn something new everyday... :bulb: